tests/browser/test_controller_download.py

```
@@ -0,0 +1,41 @@
+@pytest.mark.asyncio
+async def test_main_world_evaluate_surfaces_javascript_errors() -> None:
+    cdp = AsyncMock()
+    cdp.send.return_value = {
+        "result": {"type": "object"},
+        "exceptionDetails": {
+            "text": "Uncaught",
+            "exception": {"description": "Error: blob read failed"},
+        },
+    }
+
+    with pytest.raises(RuntimeError, match="CDP Runtime.evaluate failed: Error: blob read failed"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)